building-blocks-of-oop-part-1/src/Triangle.ts

```diff
@@ -0,0 +1,73 @@
+import { Shape } from "./Shape";
+import { Point } from "./Point";
+
+enum TriangleTypes {
+  EQUILATERAL = "equilateral triangle",
+  ISOSCELES = "isosceles triangle",
+  SCALENE = "scalene triangle",
+}
+
+type TriangleType = TriangleTypes.EQUILATERAL | TriangleTypes.ISOSCELES | TriangleTypes.SCALENE;
+
+export class Triangle extends Shape {
+  constructor(firstPoint: Point, secondPoint: Point, thirdPoint: Point);
+  constructor(firstPoint: Point, secondPoint: Point, thirdPoint: Point, color: string, filled: boolean);
```

Comment:
JFYI 

as for me - better to pass args as an argument than to have comma separated arguments in case there are more than 3 args